File: Br.Com.Fiap.Gere.Residuo/Model/AgendaModel.cs
namespace Br.Com.Fiap.Gere.Residuo.Model
{
    public class AgendaModel
    {
        
        public int AgendaId { get; set; }

        // Relacionamento com CaminhaoModel
        public int CaminhaoId { get; set; }
        public CaminhaoModel CaminhaoAlocadoParaAgenda { get; set; }
        
        // Relacionamento com MotoristaModel
        public int MotoristaId { get; set; }
        public MotoristaModel MotoristaAlocadoParaAgenda { get; set; }

        // Relacionamento com BairroModel
        public int BairroId { get; set; }
        public BairroModel BairroAgendadoParaColeta { get; set; }

        // Relacionamento com NotificacaoModel
        public NotificacaoModel NotificacaoGeradaParaEstaAgenda { get; set; }

        public DateOnly DiaColetaDeLixo { get; set; }

        public TipoResiduo TipoResiduo { get; set; }

        public StatusColetaDeLixo StatusColetaDeLixoAgendada { get; set; } = StatusColetaDeLixo.EM_ANDAMENTO;

        public long PesoColetadoDeLixoKg { get; set; } = 0;
    }
}

File: BDD_Tests/BDD-Gere-Residuo/src/test/java/model/agenda/AgendaModel.java
package model.agenda;

import com.google.gson.annotations.Expose;
import lombok.Data;
import model.enums.StatusColetaDeLixo;
import model.enums.TipoResiduo;

import java.util.Date;

@Data
public class AgendaModel {
    @Expose(serialize = false)
    private int agendaId;
    @Expose(serialize = false)
    private int pesoColetadoDeLixoKg;

    @Expose
    private int caminhaoId;
    @Expose
    private int motoristaId;
    @Expose
    private int bairroId;
    @Expose
    private String diaColetaDeLixo;
    @Expose
    private StatusColetaDeLixo statusColetaDeLixoAgendada;
    @Expose
    private TipoResiduo tipoResiduo;

}
